web/yo/app/scripts/services/importer.js

```diff
@@ -73,73 +73,103 @@ angular.module('oncokbApp')
-                                                            var record = newMetaModel.createMap();
-                                                            record.set('review', originalMeta.get(hugoSymbol).get(uuid).get('review'));
-                                                            uuidMapping.set(uuid, record);
-                                                        } else if (uuid === 'CurationQueueArticles') {
-                                                            uuidMapping.set('CurationQueueArticles', originalMeta.get(hugoSymbol).get('CurationQueueArticles'));
-                                                        } else if (uuid === 'AllArticles') {
-                                                            uuidMapping.set('AllArticles', originalMeta.get(hugoSymbol).get('AllArticles'));
-                                                        }
-                                                    });
-                                                    newReview.set(hugoSymbol, uuidMapping);
-                                                });
-                                                newMetaModel.getRoot().set('review', newReview);
-                                                console.log('Completed back up meta file');
-                                                backupGene(parentFolderId, callback);
-                                            }
-                                        });
-                                    }
-                                });
-                            }
+                // create additional data folder
+                storage.createFolder(parentFolderId, 'Additionals').then(function(folderResult) {
+                    // open up the original additional files for copying purpose
+                    additionalFile.load(['all']).then(function(result) {
+                        var newFolderId = folderResult.id;
+                        var apiCalls = [backupMeta(newFolderId), backupQueues(newFolderId)];
+                        $q.all(apiCalls).then(function(result) {
+                            backupGene(parentFolderId, callback);
+                        }, function(error) {
+                            console.log(error);
                         });
                     });
                 });
-
             });
         };
+        function backupMeta(parentFolderId) {
+            var deferred = $q.defer();
+            storage.createDocument('Meta', parentFolderId).then(function(file) {
+                console.log('Created meta file');
+                storage.getAdditionalRealtimeDocument(file.id).then(function(metaRealtime) {
+                    if (metaRealtime && metaRealtime.error) {
+                        deferred.reject('Failed to get new meta realtime document');
+                    } else {
+                        var newMetaModel = metaRealtime.getModel();
```

Comment:
I believe line 104-121 can be used to create meta model if the document does not have one. The same thing applies to backupQueues below as well. Probably then this issue can be solved? https://github.com/oncokb/oncokb/issues/781

Reply:
In this structure, Meta and Queues documents are mandatory. We can't load the home page if either one is missing. Thus by the point that we can do backuping, we are sure that meta model and queues model do exist, and no need to check. Regarding the issue #781, we only need to worry about the history model, which is solved in this PR by changes in "/databaseconnector.js"